improvement(entrance): better buttons with label

bsoufflet opened the pull request:
## 🤔 What  
- try to always add tooltip when possible on buttons
- only use icon button (with no label) for really easy to understand icon
- use button groups for some buttons which are close to each other
- easier to see "Add / New" button
- light improvements on the creator / updator component for more compact
- show the edit button as disabled (instead of hidden)
- better handle title of entrance on small screen
- organization page : use the top right default edit button
- snapshot accordion ui improvement and date fix - FIX #857 
- now using "revisions" and "revision history" instead of snapshot
- AuthorAndDate and Contribution now support with hours
- save selected language in localstorage
- new line divider before all elements in each sections of entrance page
- fix missing entrance value on entrances revision page
- fix edit button on person page
- fix restoreSnapshot
- improv ux of restore by showing a "completed" message + updating the parent page on success
- better restore button
- improve unsubscribe UI for massif + country from my account page

## 🤷‍♂️ Why  
better UI and UX

## 📸 Previews
![Screenshot from 2023-02-18 01-08-34](https://user-images.githubusercontent.com/985903/219822689-8a525fe7-4706-47e3-80ef-372d6da48de7.png)
![Screenshot from 2023-02-18 01-08-09](https://user-images.githubusercontent.com/985903/219822690-861e922c-b32d-4216-854c-e60cedd0cfdd.png)
![Screenshot from 2023-02-18 01-06-05](https://user-images.githubusercontent.com/985903/219822693-2e1128b3-6125-40f4-8af0-83e285915d7e.png)
![Screenshot from 2023-02-18 01-05-39](https://user-images.githubusercontent.com/985903/219822696-f6ed50f4-9845-47ba-82fd-5e57d269a909.png)
![Screenshot from 2023-02-18 01-05-10](https://user-images.githubusercontent.com/985903/219822697-85cbfffe-16b2-4f9c-8d15-adfeb4663fd4.png)
![Screenshot from 2023-02-21 00-09-47](https://user-images.githubusercontent.com/985903/220220932-77ae9ce3-3755-4a60-ae4c-39fd91ea2b19.png)
![Screenshot from 2023-02-21 00-09-04](https://user-images.githubusercontent.com/985903/220220934-e481f4c8-0c42-41e9-be05-602c57c9bf20.png)
![Screenshot from 2023-02-21 00-08-51](https://user-images.githubusercontent.com/985903/220220935-ea8db3d6-ec0b-4745-8c3d-bdeace2416b7.png)
![Screenshot from 2023-02-21 00-08-16](https://user-images.githubusercontent.com/985903/220220937-2f716464-a79d-4397-8f12-a1a33b5962d8.png)

bsoufflet committed:
improvement(entrance): better buttons with label

- try to always add tooltip when possible on buttons
- only use icon button (with no label) for really easy to understand icon
- use button groups for some buttons which are close to each other
- easier to see "Add / New" button
- light improvements on the creator / updator component for more compact
- show the edit button as disabled (instead of hidden)
- better handle title of entrance on small screen
- organization page : use the top right default edit button

urien commented:
C'est beau !
Est ce qu'il serait possible de faire la même chose pour le bouton qui est dans les snapshots et qui permet de revenir à un version précedente. Celui-là aussi il n'est pas très clair
![image](https://user-images.githubusercontent.com/1165144/219845923-9858ac27-d3a3-40c5-9729-2ffe71be1daf.png)

Il y a aussi une petite amélioration que je propose. 

Dans chaque carte on a le bouton nouveau qui est bien aligné avec le titre de la section mais pas les boutons snapshot et édition avec le titre de l'élément concerné.
![image](https://user-images.githubusercontent.com/1165144/219846005-9fa6d14b-195c-462b-98cd-d66640cca1ac.png)
Est ce qu'il serait possible de rajouter une ligne fine entre le titre de la section et le premier élément de cette section ? Quand il y a plusieurs éléments cela ne pose pas de pb car il y a une ligne qui permet bien d'associer les boutons avec l'élément qui est concerné
![image](https://user-images.githubusercontent.com/1165144/219846090-b7173cda-fdb2-4a01-9504-386841856c60.png)

Merci pour ce super travail

urien commented:
Sur la page des personnes il y a in bouton pour se désabonner aux notifications qu'il faudrait traiter de la même façon que sur la page des massifs et des pays
![image](https://user-images.githubusercontent.com/1165144/219846508-b1b477de-d8d9-4343-ba70-cd948a9aaf81.png)

bsoufflet committed:
chore(ci): fix build by forcing node 16

bsoufflet committed:
improvement(snapshots): better global ui

- snapshot accordion ui improvement and date fix
- now using "revisions" and "revision history" instead of snapshot
- AuthorAndDate and Contribution now support with hours
- save selected language in localstorage
- new line divider before all elements in each sections of entrance page
- fix missing entrance value on entrances revision page
- fix edit button on person page
- fix restoreSnapshot
- improv ux of restore by showing a "completed" message + updating the parent page on success
- better restore button
- improve unsubscribe UI for massif + country from my account page

urien commented:
C'est super... Mais il y a des problèmes de lien dans les révisions. Je te donne un exemple ici mais si tu es d'accord je peux faire un ticker séparé sur les révisions avec des tests systématiques
Depuis cette entrée https://orange-rock-0d4f87503-865.westeurope.azurestaticapps.net/ui/entrances/37497
Si je demande les révisions de l'historique je tombe sur https://orange-rock-0d4f87503-865.westeurope.azurestaticapps.net/ui/histories/8058/snapshots
La version actuelle  est indiqué comme étant la plus ancienne, Ce qui apparait comme la version actuelle n'est pas lié à cet historique et semble réaparaitre pour les commentaires
https://orange-rock-0d4f87503-865.westeurope.azurestaticapps.net/ui/comments/2673/snapshots
Les révisions pour la fiche d'équipement ne s'affichent pas https://orange-rock-0d4f87503-865.westeurope.azurestaticapps.net/ui/riggings/1349/snapshots

vmarseguerra commented:
@bsoufflet The code looks good, just a few suggestions on the design: 

Each comment is now very high, may be the ratings and timing can be on a single line?

![image](https://user-images.githubusercontent.com/6350275/220306374-b8647ff3-a4fa-4ae5-b50a-f7f242162fef.png)

The offset between the edit and revision buttons and the title feel weird, maybe they can be put on the same line? What do you think?

![image](https://user-images.githubusercontent.com/6350275/220305242-0799dff6-b73a-49e8-a789-54a394f8b672.png)

The last divider before the end of a card seem unnecessary, can it be removed?
This was also present before, but maybe it could also done in this PR

![image](https://user-images.githubusercontent.com/6350275/220308119-d3169892-599e-4047-a0fa-8cb3d12beb6b.png)

bsoufflet commented:
> C'est super... Mais il y a des problèmes de lien dans les révisions. Je te donne un exemple ici mais si tu es d'accord je peux faire un ticker séparé sur les révisions avec des tests systématiques Depuis cette entrée https://orange-rock-0d4f87503-865.westeurope.azurestaticapps.net/ui/entrances/37497 Si je demande les révisions de l'historique je tombe sur https://orange-rock-0d4f87503-865.westeurope.azurestaticapps.net/ui/histories/8058/snapshots La version actuelle est indiqué comme étant la plus ancienne, Ce qui apparait comme la version actuelle n'est pas lié à cet historique et semble réaparaitre pour les commentaires https://orange-rock-0d4f87503-865.westeurope.azurestaticapps.net/ui/comments/2673/snapshots Les révisions pour la fiche d'équipement ne s'affichent pas https://orange-rock-0d4f87503-865.westeurope.azurestaticapps.net/ui/riggings/1349/snapshots

@urien Je n'ai pas tous les problèmes que tu décris...
![image](https://user-images.githubusercontent.com/985903/220492592-741730ee-ddf2-40ce-bbfb-2085bae7c6c2.png)
Pour les revisions des commentaires et historique ca semble bon pour moi aussi mais par contre les dates affichées en bas à droite semble souvent pas bonne...

Si c'est une régression (problème ici mais pas sur grottocenter.org) alors je peux le traiter ici. Sinon il faudrait faire des ticket github je pense...

Il y a aussi un problème important qu'il faudra régler avec l'archi de la page des revisions : 
- On ne sais pas d'où (de quelle cavité ou entrée ou massif) l'élément pour lequel on voit les révisions est tiré. Ce n'est affiché null part et il faudrait un petit refactor back + front pour pouvoir l'afficher
- l'accès à la page de revisions directement depuis un lien (comme tu le proposes dans ton commentaire) n'est pas possible et abouti à une page cassée car l'archi actuelle faite par les étudiants utilise le local storage pour stocker l'élément parent... Il faudrait rendre cette page autonome et récupérer depuis l'API le parent plutot que de le chercher dans le local storage. Ce pattern est très mauvais il me semble... (ticket à créer)

urien commented:
J'étais partit pour te mettre des copies d'écran mais tout est ok ce matin.
Je pense que cela vient de ce que tu as décrit (local storage).
Et ce ma tin c'est super bien !
Merci Benjamin

urien commented:
Test plus complet

try to always add tooltip when possible on buttons **OK**
only use icon button (with no label) for really easy to understand icon **OK**
use button groups for some buttons which are close to each other **OK**
easier to see "Add / New" button **Bien d'avoir fait la distinction entre nouveau et ajouter**
light improvements on the creator / updator component for more compact **C'est parfait, est ce que tu pourrait ajouter la mise à jour de l'entrée sur la page de l'entrée et le nom du contributeur sur la dernière version de la révision de l'entrée : actuellement on ne sait pas qui a fait la dernière version**
show the edit button as disabled (instead of hidden) **OK**
better handle title of entrance on small screen
organization page : use the top right default edit button **OK**
snapshot accordion ui improvement and date fix - FIX **OK**
https://github.com/GrottoCenter/grottocenter-front/issues/857
now using "revisions" and "revision history" instead of snapshot **OK peut être enlever la majuscule après les : **
AuthorAndDate and Contribution now support with hours **Dans la liste des révisions : OK**
save selected language in localstorage
new line divider before all elements in each sections of entrance page **OK**
fix missing entrance value on entrances revision page **OK**
fix edit button on person page **OK**
fix restoreSnapshot **Cela fonctionne bien**
improv ux of restore by showing a "completed" message + updating the parent page on success **c'est vraiment super plus confortable : merci ! Cela n'a pas fonctionné pour une entrée dont j'avais changé de nom : pas de rafraichissement de la page). Pour la fiche d'équipement la page revient sur la page révision et pas sur la page de la cavité. Pour le document j'ai un message d'erreur quand je demande la page des révisions après avoir validé des modifications sur le document et rechargé la page de l'entrée qui est associée à ce document https://orange-rock-0d4f87503-865.westeurope.azurestaticapps.net/ui/entrances/87837**
better restore button **OK**
improve unsubscribe UI for massif + country from my account page **OK**

bsoufflet committed:
improvement(entrance): better ui and responsive

- comment more compact
- handle large rigging tables on small screen
- fix unseless last divider on all list of items
- document more compact and responsive

bsoufflet commented:
@urien : Je te propose de régler les derniers points qui ne sont pas ok dans des tickets séparés afin de ne pas faire une PR à rallonge ;)

urien commented:
Oui, c'est bien mieux de faire ainsi. 
Je refais des tests et ouvre un ticket
